scripts/cvmfsexec-osg-wrapper

```diff
@@ -114,30 +114,62 @@ if [[ -n $CVMFS_HTTP_PROXY ]]; then
     add_or_replace "$cvmfs_local_config" CVMFS_HTTP_PROXY "${CVMFS_HTTP_PROXY}"
 fi
 
-"$workdir"/cvmfsexec/umountrepo -a || :
-
-trap '"$workdir"/cvmfsexec/umountrepo -a; cd "$prevdir"; rm -rf "$workdir"' EXIT
+# If we have to use this script to mount CVMFS, we can't use images from there anyway.
+export ALLOW_NONCVMFS_IMAGES=true
 
-# Mount repos
-for repo in config-osg.opensciencegrid.org $CVMFSEXEC_REPOS; do
-    "$workdir"/cvmfsexec/mountrepo "$repo" || fail "Unable to mount cvmfs repo $repo"
-done
+echo >&2 "*** Running cvmfsexec smoke test:"
+if "$workdir"/cvmfsexec/cvmfsexec -N -- /bin/true; then
+    echo >&2 "*** cvmfsexec smoke test passed: you have the permissions to run cvmfsexec directly"
 
-export CVMFS_BASE="$workdir"/cvmfsexec/dist/cvmfs
+    # cvmfsexec lets us have CVMFS mounted at `/cvmfs`; set $CVMFS_BASE for
+    # consistency with the mountrepo path
 
-export GLIDEIN_SINGULARITY_BINDPATH="$CVMFS_BASE:/cvmfs"
+    export CVMFS_BASE=/cvmfs
+    export GLIDEIN_SINGULARITY_BINDPATH="$CVMFS_BASE:/cvmfs"
+    echo "*** cvmfs available at $CVMFS_BASE" >&2
 
-# If we have to use this script to mount CVMFS, we can't use images from there anyway.
-export ALLOW_NONCVMFS_IMAGES=true
+    # Note: do not use 'exec'; cleanup won't run
+    "$workdir"/cvmfsexec/cvmfsexec -N $CVMFSEXEC_REPOS -- "$@"
```

Comment:
```suggestion
    "$workdir"/cvmfsexec/cvmfsexec -N "$CVMFSEXEC_REPOS" -- "$@"
```

This comes from the user, right?

Reply:
Correct and it _should_ be expanded into separate arguments. It's not eval'ed so we don't have to worry about "little Bobby Tables" issues, although I could sanitize the variable names for safety's sake...